src/kixi/event_replay.clj

```diff
@@ -0,0 +1,176 @@
+            [kixi.event-replay.inputstream-to-events :refer [input-stream->events]]))
+
+(def config (atom {}))
+
+(def datehour-formatter
+  (f/formatter :date-hour))
+
+(def parse-datahour
+  (partial f/parse datehour-formatter))
+
+(def unparse-datahour
+  (partial f/unparse datehour-formatter))
+
+(defn datehour?
+  [x]
+  (if (instance? org.joda.time.DateTime x)
+    x
+    (try
+      (if (string? x)
+        (parse-datahour x)
+        :clojure.spec.alpha/invalid)
+      (catch IllegalArgumentException e
+        :clojure.spec.alpha/invalid))))
+
+(def datehour
+  (s/conformer datehour? identity))
+
+(s/def ::start-datehour datehour)
+(s/def ::end-datehour 
+  (s/or :t datehour
+        :n nil?))
+
+(s/def ::target-stream string?)
+(s/def ::s3-base-dir
+  (s/and string?
+         #(string/ends-with? % "-backup")))
+
+(s/def ::config
+  (s/keys :req-un [::start-datehour
+                   ::end-datehour
+                   ::target-stream
+                   ::s3-base-dir]))
+
+(defn validate-config
+  [config]
+  (if (s/valid? ::config config)
+    [true (s/conform ::config config)]
+    [false (s/explain-data ::config config)]))
+
+(def one-hour (t/hours 1))
+
+(defn hour-sequence
+  []
+  (let [{:keys [start-datehour
+                end-datehour]} @config
+        end-datehour (or end-datehour 
+                         (unparse-datahour (t/now)))
+        start (parse-datahour start-datehour)
+        end (parse-datahour end-datehour)]    
+    (p/periodic-seq start
+                    (t/plus end
+                            one-hour)
+                    one-hour)))
+
+(defn hour->s3-prefix
+  [hour]
+  (->> [(t/year hour)       
+        (t/month hour)
+        (t/day hour)
+        (t/hour hour)]
+       (map str)
+       (map #(if (= 1 (count %))
+               (str "0" %)
+               %))
+       (interpose "/")
+       (apply str)))
+
+(def max-objects 20)
+
+(defn hour->s3-object-summaries
+  ([hour]
+   (hour->s3-object-summaries (:s3-base-dir @config) 
+                              (hour->s3-prefix hour)
+                              nil))
+  ([s3-base-dir prefix marker]
+   (let [list-objects-res (s3/list-objects (merge {:bucket-name s3-base-dir
+                                                   :prefix prefix
+                                                   :max-keys max-objects}
+                                                  (when marker
+                                                    {:marker marker})))]
+     (concat (:object-summaries list-objects-res)
+             (when (:next-marker list-objects-res)
+               (hour->s3-object-summaries s3-base-dir prefix (:next-marker list-objects-res)))))))
+
+(defn object-summary->input-stream
+  [s3-object-summary]
+  (let [{:keys [s3-base-dir
+                target-stream]} @config
+        s3-object (s3/get-object :bucket-name s3-base-dir
+                                 :key (:key s3-object-summary))]
+    (:object-content s3-object)))
```

Comment:
Should this not be `:input-stream`?

Reply:
Sadly not, the object stream is an input stream of the objects contents